### PR TITLE
Support custom DN entries

### PR DESCRIPTION
**Pull request by randombit**
GH #1490

### CODE REVIEW

File: src/lib/x509/x509_dn.cpp [hunk 1]
```diff
@@ -201,58 +210,32 @@ bool operator<(const X509_DN& dn1, const X509_DN& dn2)
...
    {
-   auto dn_info = get_attributes();
-
    der.start_cons(SEQUENCE);
 
    if(!m_dn_bits.empty())
+      {
+      /*
+      If we decoded this from somewhere, encode it back exactly as
+      we recieved it
```
Comment (phwork) [inline src/lib/x509/x509_dn.cpp]: `received` ;)

File: src/lib/x509/x509_dn.cpp [hunk 2]
```diff
@@ -201,58 +210,32 @@ bool operator<(const X509_DN& dn1, const X509_DN& dn2)
...
    else
       {
-      do_ava(der, dn_info, PRINTABLE_STRING, "X520.Country");
-      do_ava(der, dn_info, DIRECTORY_STRING, "X520.State");
-      do_ava(der, dn_info, DIRECTORY_STRING, "X520.Locality");
-      do_ava(der, dn_info, DIRECTORY_STRING, "X520.Organization");
-      do_ava(der, dn_info, DIRECTORY_STRING, "X520.OrganizationalUnit");
-      do_ava(der, dn_info, DIRECTORY_STRING, "X520.CommonName");
-      do_ava(der, dn_info, PRINTABLE_STRING, "X520.SerialNumber");
+      for(auto dn : m_dn_info)
```
Comment (phwork) [inline src/lib/x509/x509_dn.cpp]: could use `const auto&`

**Comment by randombit** [comment]
@phwork thank you kindly for the review

Paging also @neusdan + @securitykernel - this PR changes the ordering that DN entries are encoded in. Previously the order was specified by the sequence of calls to `do_ava`. Now we encode by OID ordering. Thus the changes to the DN hash test. I don't think there are any troublesome implications to this but wanted to run it by you.

**Comment by neusdan** [comment]
LGTM!

**Pull request merged**